controllers/humiocluster_controller.go

```diff
@@ -1168,6 +1173,74 @@ func (r *HumioClusterReconciler) ensurePvcLabels(ctx context.Context, hc *humiov
 	return nil
 }
 
+func (r *HumioClusterReconciler) ensureLicense(ctx context.Context, hc *humiov1alpha1.HumioCluster) (reconcile.Result, error) {
+	r.Log.Info("ensuring license")
+
+	existingLicense, err := r.HumioClient.GetLicense()
+	if err != nil {
+		r.Log.Info(fmt.Sprintf("failed to get license: %v", err))
+	}
+
+	defer func(ctx context.Context, hc *humiov1alpha1.HumioCluster) {
+		licenseStatus := humiov1alpha1.HumioLicenseStatus{
+			Type:       existingLicense.LicenseType(),
+			Expiration: existingLicense.ExpiresAt(),
+		}
+		r.setLicense(ctx, licenseStatus, hc)
+	}(ctx, hc)
+
+	licenseSecretKeySelector := licenseSecretKeyRefOrDefault(hc)
+	if licenseSecretKeySelector == nil {
+		return reconcile.Result{}, nil
+	}
+
+	licenseSecret, err := kubernetes.GetSecret(ctx, r, licenseSecretKeySelector.Name, hc.Namespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Error(err, fmt.Sprintf("license was requested but no secret exists by name %s in namespace %s", licenseSecretKeySelector.Name, hc.Namespace))
+			return reconcile.Result{}, err
+		}
+	}
```

Comment:
Looks like we right now ignore any errors that are not true when calling `errors.IsNotFound()`. Is that intentional?